libsolidity/formal/SMTChecker.cpp

```diff
@@ -273,6 +273,23 @@ void SMTChecker::endVisit(VariableDeclarationStatement const& _varDecl)
 		);
 }
 
+bool SMTChecker::visit(Assignment const& _assignment)
+{
+	// Erase knowledge about references before the right hand side smt::Expression is created
```

Comment:
Can you explain that? What if the LHS is used in the RHS? Do you have an example?

Reply:
```
b[2] = y;
a[2] = x;
```
In this case, after the second assignment, knowledge about `b` should be cleared since `a = b` is possible. Without the code below knowledge about `a` would also be cleared. But now that I think about it I think the current code would not clear `a`, only the other variables. Let me check.

Reply:
Yes, this function is not necessary anymore, removed.